### ISSUE TITLE
Error with define if it uses variable in name.

### ISSUE THREAD

**Issue by marcimat** [phpDocumentor/Reflection#237]
Hi, first of all thanks for PHPDocumentor version 3.3.1-rc which correct #233. 
Now I can see another singular issue with constant name when defined like that : `define('NOT_OK_' . $v, 'not ok');`

### Composer.json
```json
{
    "require": {
        "phpdocumentor/reflection": "5.x-dev#3b54c5be5143f5a1e002c3309b27f6af9eb33abf"
    }
}
```

### test.php
```php
<?php
$v = '1';
define('NOT_OK_' . $v, 'not ok');
#echo NOT_OK_1;
```

### run.php
```php
<?php
ini_set('display_errors', 'on');
error_reporting(E_ALL);
include_once 'vendor/autoload.php';
$projectFactory = \phpDocumentor\Reflection\Php\ProjectFactory::createInstance();
$projectFiles = [new \phpDocumentor\Reflection\File\LocalFile('test.php')];
$project = $projectFactory->create('My Project', $projectFiles);
var_dump($project);
```

### execute
```
php demo.php
```

### gives
```php
Fatal error: Uncaught AssertionError: assert($nameString instanceof String_) in ~/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Php/Factory/Define.php:128
```

Thanks a lot :)

**Comment by jaapio** [comment]
Hi,

Thanks for your report, The easiest way to fix this would be to ignore dynamic defines. As this library is performing a simple code parsing and not a dynamic analysis of the code, we will not be able to resolve the correct naming for those defines. So I think the best we can get for now is skipping them without an error. Does that make sense to you?

**Comment by jaapio** [comment]
relates to #182

**Comment by marcimat** [comment]
That seems good to me. Better than fatal error :)

**Comment by jaapio** [comment]
This should have been fixed in v5.2.0
